src/universal/helpers/bag.test.ts

```diff
@@ -0,0 +1,77 @@
+import { BAGSourceData } from '../types/bag';
+import {
+  extractAddress,
+  getBagSearchAddress,
+  getLatLonByAddress,
+  isLocatedInWeesp,
+} from './bag';
+
+describe('getLatLonByAddress', () => {
+  const weesp = 'Herengracht 23';
+  const amsterdam = 'Herengracht 23-1';
+
+  const response: BAGSourceData = {
+    results: [
+      {
+        adres: 'Herengracht 23-1',
+        centroid: [4.891968036478453, 52.37873183842775],
+        woonplaats: 'Amsterdam',
+      },
+      {
+        adres: 'Herengracht 23-2',
+        centroid: [4.891968036478453, 52.37873183842775],
+        woonplaats: 'Amsterdam',
+      },
+      {
+        adres: 'Herengracht 23-H',
+        centroid: [4.891968036478453, 52.37873183842775],
+        woonplaats: 'Amsterdam',
+      },
+      {
+        adres: 'Herengracht 23',
+        centroid: [5.039817231849981, 52.30885683238395],
+        woonplaats: 'Weesp',
+      },
+      {
+        adres: 'Herengracht 231a',
+        centroid: [5.03863916842061, 52.30886128545404],
+        woonplaats: 'Weesp',
+      },
+      {
+        adres: 'Nieuwe Herengracht 23-1',
+        centroid: [4.902795334609859, 52.36631966746123],
+        woonplaats: 'Amsterdam',
+      },
+    ],
+  };
+
+  test('Amsterdam', () => {
+    expect(getLatLonByAddress(response?.results, amsterdam, false)).toEqual({
+      lat: 52.37873183842775,
+      lng: 4.891968036478453,
+    });
+  });
+
+  test('Weesp', () => {
+    expect(getLatLonByAddress(response?.results, weesp, true)).toEqual({
+      lat: 52.30885683238395,
+      lng: 5.039817231849981,
+    });
+  });
+
+  test('extractAddress', () => {
+    expect(extractAddress('Herengracht 23-1, 1015BA, Amsterdam _ ; . ,')).toBe(
+      'Herengracht 23-1'
+    );
+  });
+
+  test('isWeesp', () => {
+    expect(isLocatedInWeesp('Weesperstraat 113 Amsterdam')).toBe(false);
```

Comment:
Zou je hier ook niet een test willen met een adres in Weesp om te controleren of de functie dan true teruggeeft.